generator/lib/src/generator.dart

```diff
@@ -1732,21 +1732,29 @@ ${bodyName.displayName} == null
         final contentType = r.peek('contentType')?.stringValue;
 
         if (isFileField) {
+          final keepFilePath = r.peek('keepFilePath')?.boolValue ?? false;
           final fileNameValue = r.peek('fileName')?.stringValue;
           final fileName = fileNameValue != null
               ? literalString(fileNameValue)
               : refer(p.displayName)
                   .property('path.split(Platform.pathSeparator).last');
+          final filePath = refer(p.displayName).property('path');
+          final headers = keepFilePath
+              ? literalMap({
+                  'original_file_path': [filePath]
```

Comment:
`literalList` is needed here to return a list in generated code

Reply:
if not,  remove square brackets .

Reply:
We need the list here, else we get an error once we build the file. Thanks, I've added the literalList